lightningd/options.c

```diff
@@ -1213,6 +1218,72 @@ static char *opt_set_announce_dns(const char *optarg, struct lightningd *ld)
 	return opt_set_bool_arg(optarg, &ld->announce_dns);
 }
 
+static char *opt_set_codex32(const char *arg, struct lightningd *ld)
+{
+	struct codex32_parts *parts = tal(tmpctx, struct codex32_parts);
+	const uint8_t *payload;
+	char *err;
+
+	/* Checks if hsm_secret exists */
+	int fd = open("hsm_secret", O_CREAT|O_EXCL|O_WRONLY, 0400);
+	if (fd < 0) {
+		/* Don't do anything if the file already exists. */
+		if (errno == EEXIST)
+			return tal_fmt(tmpctx, "HSM_Secret already exists!");
+	}
+
```

Comment:
and otherwise, you need to still return an error, so "return tal_fmt(tmpctx, "Creating hsm_secret: %s",  strerror(errno));"

Reply:
Good catch, Thanks!